File: week1/AhmetTahaSener/TwoSum.cs
public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        int[] result = new int[2];
        for (int i = 0; i < nums.Length; i++)
        {
            if (nums.Contains(target - nums[i]))
            {
                result[0] = i;
                result[1] = Array.IndexOf(nums, target - nums[i]);
                if (!(result[0] == result[1]))
                {
                    return result;
                }
                else
                {
                    continue;
                }
            }
        }
        return result;
    }
}

File: week1/omerSefaYilmaz/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

var twoSum = function (nums, target) {
    let remainPairs = {};
    let result = [];
    for (let i = 0; i < nums.length; i++) {
      if (remainPairs.hasOwnProperty([nums[i]])) {
        result.push(i, nums.indexOf(remainPairs[nums[i]]));
      } else {
        remainPairs[target - nums[i]] = nums[i];
      }
    }
    return result;
  };